### PR TITLE
SyncBot: Move metadata to end of message

### PR DESCRIPTION
**Pull request by sqweelygig**
* Improves rendering of comments in Front
* Improves integration with Hubot

Change-Type: patch
Connects-To: #343 

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [x] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #343

### CODE REVIEW

File: lib/services/messenger/translators/translator-scaffold.ts [hunk 1]
```diff
@@ -74,10 +74,10 @@ export abstract class TranslatorScaffold implements Translator {
 				const charRegex = new RegExp(`^${charCapture}`, 'im');
 				return TranslatorScaffold.metadataByRegex(message, charRegex);
 			case MetadataEncoding.HiddenHTML:
-				const hiddenHTMLRegex = new RegExp(`^<a href="${baseUrl}${querystring}"[^>]*></a>`, 'im');
+				const hiddenHTMLRegex = new RegExp(`<a href="${baseUrl}${querystring}"[^>]*></a>`, 'im');
 				return TranslatorScaffold.metadataByRegex(message, hiddenHTMLRegex);
 			case MetadataEncoding.HiddenMD:
-				const hiddenMDRegex = new RegExp(`^\\[\\]\\(${baseUrl}${querystring}\\) `, 'im');
+				const hiddenMDRegex = new RegExp(`\\[\\]\\(${baseUrl}${querystring}\\)`, 'im');
```
Comment (pimterry) [inline lib/services/messenger/translators/translator-scaffold.ts]: These two mean now that I can post a message and (accidentally/intentionally) include some text that's interpreted as metadata in my own message. Does that have security/awkward bug repercussions?

Comment (sqweelygig) [reply lib/services/messenger/translators/translator-scaffold.ts]: Okay, so the email that Front emits (checked against the 'show original'), does not simply end with </a>, so we can be much more discerning by checking that this is at the actual end of the string.  Just committing this wisdom now.

File: lib/services/messenger/translators/flowdock.ts [hunk 2]
```diff
@@ -50,12 +50,12 @@ export class FlowdockTranslator extends TranslatorScaffold implements Translator
 		const inject = metadata ? metadata : '';
 		const last = footer ? `\n${footer}` : '';
 		if ((first.length + inject.length + body.length + last.length) < lengthLimit) {
-			return `${first}${inject}${body}${last}`;
+			return `${first}${body}${last}${inject}`;
```
Comment (LucianBuzzo) [inline lib/services/messenger/translators/flowdock.ts]: This is the perfect opportunity to add some unit tests for this pure function :)

File: lib/services/messenger/translators/translator-scaffold.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ export abstract class TranslatorScaffold implements Translator {
 			case MetadataEncoding.HiddenHTML:
 				return `<a href="${process.env.MESSAGE_TRANSLATOR_ANCHOR_BASE_URL}${queryString}"></a>`;
 			case MetadataEncoding.HiddenMD:
-				return `[](${process.env.MESSAGE_TRANSLATOR_ANCHOR_BASE_URL}${queryString}) `;
+				return `[](${process.env.MESSAGE_TRANSLATOR_ANCHOR_BASE_URL}${queryString})`;
```
Comment (LucianBuzzo) [inline lib/services/messenger/translators/translator-scaffold.ts]: Unit tessssts

File: lib/services/messenger/translators/translator-scaffold.ts [hunk 4]
```diff
@@ -74,10 +74,10 @@ export abstract class TranslatorScaffold implements Translator {
 				const charRegex = new RegExp(`^${charCapture}`, 'im');
 				return TranslatorScaffold.metadataByRegex(message, charRegex);
 			case MetadataEncoding.HiddenHTML:
-				const hiddenHTMLRegex = new RegExp(`^<a href="${baseUrl}${querystring}"[^>]*></a>`, 'im');
+				const hiddenHTMLRegex = new RegExp(`<a href="${baseUrl}${querystring}"[^>]*></a>$`, 'i');
```
Comment (LucianBuzzo) [inline lib/services/messenger/translators/translator-scaffold.ts]: Unit tests to make sure we never have regressions or security issues in the future :)

**Comment by sqweelygig** [comment]
@LucianBuzzo there was a month or two where every single one of @pimterry's reviews of my code was "semver" and then after that all of @hedss reviews were "documentation"; I sense that we may be entering a festive period of "unit tests".

File: lib/services/messenger/translators/flowdock.ts (hunk 2 above)
Comment (sqweelygig) [reply lib/services/messenger/translators/flowdock.ts]: Done

File: lib/services/messenger/translators/translator-scaffold.ts (hunk 3 above)
Comment (sqweelygig) [reply lib/services/messenger/translators/translator-scaffold.ts]: Done

File: lib/services/messenger/translators/translator-scaffold.ts (hunk 4 above)
Comment (sqweelygig) [reply lib/services/messenger/translators/translator-scaffold.ts]: Done

**Pull request merged**